server/chat/backend/agent/workflow.py

```
@@ -1743,6 +1747,55 @@ def _append_new_turn_ui_messages(
             logger.error(f"Error appending UI messages: {e}")
             return False
 
+    def _redact_for_ui(self, content: Any, tool_name: str = "") -> str:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `content`

(ANN401)